btf/btf.go

```diff
@@ -500,7 +496,14 @@ func (sw sliceWriter) Write(p []byte) (int, error) {
 // Returns an error wrapping ErrNotFound if a Type with the given ID
 // does not exist in the Spec.
 func (s *Spec) TypeByID(id TypeID) (Type, error) {
```

Comment:
From the commit message:

> TypeByID currently assumes that the first ID in s.types is 0, which is not correct when dealing with split BTF.

Could you add some context (in comments, ideally) about why this is the case? I'm not so familiar with split BTF yet and it's been a while since we touched it.

Split BTF IDs start at `len(vmlinux) + 1`, but vmlinux always acts as a base. Can we have a Spec containing only split BTF without a base?

Reply:
Right, makes sense.

> Otherwise it's not possible to query a spec to find out whether a type is in in the split or the base part.

Why is this important again? Any real-world application?

Reply:
Yes, it's required for kmod freplace.